test/unit/transform/ferc1_test.py

```diff
@@ -1 +1,84 @@
 """Unit tests specific to the FERC Form 1 table transformations."""
+
+from io import StringIO
+
+import pandas as pd
+import pytest
+
+from pudl.settings import Ferc1Settings
+from pudl.transform.ferc1 import fill_dbf_to_xbrl_map, read_dbf_to_xbrl_map
+
+TEST_DBF_XBRL_MAP = pd.read_csv(
+    StringIO(
+        """
+sched_column_name,report_year,row_literal,row_number,row_type,xbrl_column_stem
+test_table1,2000,"Header 1",1,header,"N/A"
+test_table1,2000,"Account A",2,ferc_account,account_a
+test_table1,2000,"Account B",3,ferc_account,account_b
+test_table1,2000,"Header 2",4,header,"N/A"
+test_table1,2000,"Account C",5,ferc_account,account_c
+test_table1,2002,"Header 1",1,header,"N/A"
+test_table1,2002,"Account A",2,ferc_account,account_a
+test_table1,2002,"Account B",3,ferc_account,account_b
+test_table1,2002,"Account B1",4,ferc_account,account_b1
+test_table1,2002,"Header 2",5,header,"N/A"
+test_table1,2002,"Account C",6,ferc_account,account_c
```

Comment:
@cmgosnell I changed the test such that it has 2 headers in both 2000 and 2002, but the header moves due to the insertion of Account B1. It exhibited the same bug, but I think clarifies the issue you were asking about and makes it clear that it's the displacement of row numbers which *moves* headers, and not the addition of headers, that causes the problem.

Reply:
what I was confused about is that in this example the only account that changes row number (that i can see) is `account_c` and the mechanism for ensuring that the ffil doesn't go gangbusters is the existence of  that 2002 `header` row.

I think what I now better understand about this is that if we added:
```
test_table1,2000,"Account D",6,ferc_account,account_d
...
test_table1,2002,"Account D",7,ferc_account,account_d
```
Account D's ffil would be stopped by the 2002 `account_c` record.

Reply:
Yes, because the `ffill` is only happening within a particular row-number group. So it's just filling in the definitions for e.g. row_number == 6 for all years, and whenever there's a new definition for what row 6 means, that one takes over in the ffill.

Reply:
@cmgosnell if this still isn't totally clear maybe it needs more documentation in the function itself.